python/pyarrow/tests/test_scalars.py

```diff
@@ -244,14 +244,32 @@ def test_time_from_datetime_time():
 @pytest.mark.parametrize(['value', 'time_type'], [
     (1, pa.time32("s")),
     (2**30, pa.time32("s")),
+    (None, pa.time32("s")),
     (1, pa.time32("ms")),
     (2**30, pa.time32("ms")),
+    (None, pa.time32("ms")),
     (1, pa.time64("us")),
     (2**62, pa.time64("us")),
+    (None, pa.time64("us")),
     (1, pa.time64("ns")),
     (2**62, pa.time64("ns")),
+    (None, pa.time64("ns")),
+    (1, pa.date32()),
+    (2**30, pa.date32()),
+    (None, pa.date32()),
+    (1, pa.date64()),
+    (2**62, pa.date64()),
+    (None, pa.date64()),
+    (1, pa.timestamp("ns")),
+    (2**62, pa.timestamp("ns")),
+    (None, pa.timestamp("ns")),
+    (1, pa.duration("ns")),
+    (2**62, pa.duration("ns")),
+    (None, pa.duration("ns")),
+    ((1, 2, -3), pa.month_day_nano_interval()),
+    (None, pa.month_day_nano_interval()),
 ])
-def test_time_values(value, time_type):
+def test_logical_types_values(value, time_type: pa.DataType):
```

Comment:
Logical types do not really exist in Arrow (in the sense that all types exist in the same category).
```suggestion
def test_temporal_values(value, time_type: pa.DataType):
```